lib/fabrik/DirectorManager.js

```diff
@@ -186,6 +188,21 @@ class DirectorManager extends BaseManager {
           return;
         }
       })
+      .then(() => {
+        const username = _.get(args, 'parameters.username');
+        const password = _.get(args, 'parameters.password');
+        if (action === 'create' && username !== undefined) {
```

Comment:
This check is not done in case of update & this is currently buggy, because for an update operation user can pass bosh name and it would skip this validation and endup creating a new deployment on a new bosh (potentially). Need to further fine tune these checks. May be you can do the following instead:
@line 178 - within update case switch block, you can check if username/password params are passed and if they are then throw exception 'BadRequest'
@line 188 - Move existing code from 192 - 204

Reply:
Done